### PR TITLE
Show install and activate progress around recipe solves

### PR DESCRIPTION
**Pull request by Naatan**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-501" title="DX-501" target="_blank">StoryDX-501</a>  State install has progress animations and status messages during long non-interactive phases (solving, name resolution, etc.)
  </td></table>
  <br />

### CODE REVIEW

File: cmd/state-svc/internal/rtwatcher/watcher.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ type Watcher struct {
 }
 
 func New(cfg *config.Instance, an *sync.Client) *Watcher {
-	w := &Watcher{an: an, stop: make(chan struct{}), cfg: cfg, interval: defaultInterval}
+	w := &Watcher{an: an, stop: make(chan struct{}, 1), cfg: cfg, interval: defaultInterval}
```
Comment (MDrakos) [inline cmd/state-svc/internal/rtwatcher/watcher.go:32]: nit: Is this buffer on the channel necessary? We could potentially remove the buffer and simply close the channel in the `Close()` function. Same for the `stop` channel in the `DotProgress`

File: internal/runbits/progressbar/decor.go [hunk 2]
```diff
@@ -18,6 +18,20 @@ func (pb *RuntimeProgress) addTotalBar(name string, total int64) *mpb.Bar {
 	return pb.addBar(name, total, false, mpb.BarFillerClearOnComplete())
 }
 
+// addTotalBar adds a bar counting a number of sub-events adding up to total
```
Comment (MDrakos) [inline internal/runbits/progressbar/decor.go:21]:
```suggestion
// addSpinnerBar adds a bar with a spinning progress indicator
```

File: internal/runners/packages/packages.go [hunk 3]
```diff
@@ -80,15 +95,40 @@ func executePackageOperation(prime primeable, packageName, packageVersion string
...
+		if len(packages) == 0 {
+			suggestions, err := getSuggestions(ns, packageName)
+			if err != nil {
+				logging.Error("Failed to retrieve suggestions: %v", err)
+			}
+			if len(suggestions) == 0 {
+				return locale.WrapInputError(err, "package_ingredient_alternatives_nosuggest", "", packageName)
+			}
+			return locale.WrapInputError(err, "package_ingredient_alternatives", "", packageName, strings.Join(suggestions, "\n"))
+		}
```
Comment (MDrakos) [inline internal/runners/packages/packages.go:122]: Isn't this https://activestatef.atlassian.net/browse/DX-500?

**Review by MDrakos — APPROVED** [review]
Looks good to me. Just nits and docstring updates

File: cmd/state-svc/internal/rtwatcher/watcher.go (hunk 1 above)
Comment (Naatan) [reply cmd/state-svc/internal/rtwatcher/watcher.go:32]: It's effectively a failsafe. I don't think it's necessary with how we're using this channel, but it's better to be explicit.

File: internal/runners/packages/packages.go (hunk 3 above)
Comment (Naatan) [reply internal/runners/packages/packages.go:122]: It is. I didn't introduce this code in this PR, I just moved it.

**Pull request merged**